grants_tagger/__main__.py

```diff
@@ -72,33 +103,65 @@ def train(
 
 @preprocess_app.command()
 def bioasq_mesh(
-        input_path: Optional[Path] = typer.Argument(None, help="path to BioASQ JSON data"),
-        output_path: Optional[Path] = typer.Argument(None, help="path to output JSONL data"),
-        mesh_metadata_path: Optional[Path] = typer.Option(None, help="path to xml file containing MeSH taxonomy"),
+        input_path: Optional[str] = typer.Argument(None, help="path to BioASQ JSON data"),
+        output_path: Optional[str] = typer.Argument(None, help="path to output JSONL data"),
+        mesh_metadata_path: Optional[str] = typer.Option(None, help="path to xml file containing MeSH taxonomy"),
+        filter_tags: Optional[str] = typer.Option(None, help="filter mesh subbranch like disease"),
+        test_split: Optional[float] = typer.Option(None, help="split percentage for test data. if None no split."),
         config: Optional[Path] = typer.Option(None, help="path to config files that defines arguments")):
 
+    params_path = os.path.join(os.path.dirname(__file__), "../params.yaml")
+    with open(params_path) as f:
+        params = yaml.safe_load(f)
+
+    # Default values from params
+    if not filter_tags:
+        filter_tags = params["preprocess_bioasq_mesh"]["filter_tags"]
+
     if config:
         cfg = configparser.ConfigParser()
         cfg.read(config)
 
         input_path = cfg["preprocess"]["input"]
         output_path = cfg["preprocess"]["output"]
         mesh_metadata_path = cfg["filter_disease_codes"]["mesh_descriptions_file"]
-
-    if os.path.exists(output_path):
+        filter_tags = cfg["filter_disease_codes"].get("filter_tags")
+        test_split = cfg["preprocess"].get("test_split")
+
+    # TODO: Refactor with preprocess_mesh
+    if test_split:
+        data_dir, data_name = os.path.split(output_data_path)
```

Comment:
should output_data_path be output_path?

Reply:
good catch, do not know how you find them and not sure how it passed the tests. aaa I know, I am not testing the cli very thoroughly, will add a test and fix

Reply:
Fixes here 347b64b but not adding a test now as I need some fake data for the xml which is a bit more work, will do in the future